hadoop-ozone/common/src/test/java/org/apache/hadoop/ozone/om/protocolPB/TestS3GrpcOmTransport.java

```diff
@@ -0,0 +1,129 @@
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(TestS3GrpcOmTransport.class);
+
+  private final OMResponse omResponse = OMResponse.newBuilder()
+                  .setSuccess(true)
+                  .setStatus(Status.OK)
+                  .setCmdType(Type.AllocateBlock)
+                  .build();
+
+  private final OzoneManagerServiceGrpc.OzoneManagerServiceImplBase
+      serviceImpl =
+        mock(OzoneManagerServiceGrpc.OzoneManagerServiceImplBase.class,
+            delegatesTo(
+              new OzoneManagerServiceGrpc.OzoneManagerServiceImplBase() {
+                @Override
+                public void submitRequest(org.apache.hadoop.ozone.protocol.proto
+                                              .OzoneManagerProtocolProtos
+                                              .OMRequest request,
+                                          io.grpc.stub.StreamObserver<org.apache
+                                              .hadoop.ozone.protocol.proto
+                                              .OzoneManagerProtocolProtos
+                                              .OMResponse>
+                                          responseObserver) {
+                  LOG.info("GrpcOzoneManagerServer: "
+                      + "OzoneManagerServiceImplBase "
+                      + "processing s3g client submit request");
+
+                  responseObserver.onNext(omResponse);
+                  responseObserver.onCompleted();
+                }
+
+              }));
+
+  private GrpcOmTransport client;
+
+  @Before
+  public void setUp() throws Exception {
+    // Generate a unique in-process server name.
+    String serverName = InProcessServerBuilder.generateName();
+
+    // Create a server, add service, start,
+    // and register for automatic graceful shutdown.
+    grpcCleanup.register(InProcessServerBuilder
+        .forName(serverName)
+        .directExecutor()
+        .addService(serviceImpl)
+        .build()
+        .start());
+
+    // Create a client channel and register for automatic graceful shutdown.
+    ManagedChannel channel = grpcCleanup.register(
+        InProcessChannelBuilder.forName(serverName).directExecutor().build());
+
+    String omServiceId = "";
+    OzoneConfiguration conf = new OzoneConfiguration();
+    UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
+    client = new GrpcOmTransport(conf, ugi, omServiceId);
+    client.startClient(channel);
+  }
+
+  @Test
+  public void testSubmitRequestToServer() throws Exception {
+    ServiceListRequest req = ServiceListRequest.newBuilder().build();
+
+    final OMRequest omRequest = OMRequest.newBuilder()
+        .setCmdType(Type.ServiceList)
+        .setVersion(CURRENT_VERSION)
+        .setClientId("test")
+        .setServiceListRequest(req)
+        .build();
+
+    final OMResponse resp = client.submitRequest(omRequest);
+    Assert.assertEquals(resp.getStatus(), OK);
+  }
```

Comment:
Minor Nit: Good to check custom values that are checked.
```suggestion
    Assert.assertEquals(resp.getStatus(), OK);
    Assert.assertEquals(resp.getLeaderOMNodeId(),"testOM");
  }
```

Reply:
Added custom value as noted to latest commit.  Same as suggested, using a constant to reflect OMResponse `leaderOMNodeId`.

`private final String leaderOMNodeId = "TestOM";`